gnomad_qc/v4/sample_qc/assign_ancestry.py

```diff
@@ -367,60 +557,84 @@ def write_pca_results(
+            )
+        elif args.max_proportion_mislabeled_training_samples is not None:
+            max_mislabeled = args.max_proportion_mislabeled_training_samples
+        elif args.max_number_mislabeled_training_samples is not None:
+            max_mislabeled = args.max_number_mislabeled_training_samples
+        else:
+            max_mislabeled = None
+
+        if args.run_pca:
+            pop_eigenvalues, pop_scores_ht, pop_loadings_ht = run_pca(
+                pca_related_samples_to_drop().ht(),
+                include_unreleasable_samples,
+                args.n_pcs,
+                test,
+            )
 
-    include_unreleasable_samples = args.include_unreleasable_samples
-    overwrite = args.overwrite
-    test = args.test
-    only_train_on_hgdp_tgp = args.only_train_on_hgdp_tgp
-
-    if args.run_pca:
-        pop_eigenvalues, pop_scores_ht, pop_loadings_ht = run_pca(
-            pca_related_samples_to_drop().ht(),
-            include_unreleasable_samples,
-            args.n_pcs,
-            test,
-        )
+            write_pca_results(
+                pop_eigenvalues,
+                pop_scores_ht,
+                pop_loadings_ht,
+                overwrite,
+                include_unreleasable_samples,
+                test,
+            )
 
-        write_pca_results(
-            pop_eigenvalues,
-            pop_scores_ht,
-            pop_loadings_ht,
-            overwrite,
-            include_unreleasable_samples,
-            test,
-        )
+        if args.assign_pops:
+            pop_pcs = args.pop_pcs
+            pop_pcs = list(range(1, pop_pcs[0] + 1)) if len(pop_pcs) == 1 else pop_pcs
+            logger.info("Using following PCs: %s", pop_pcs)
+            pop_ht, pops_rf_model = assign_pops(
+                args.min_pop_prob,
+                include_unreleasable_samples,
+                max_number_mislabeled_training_samples=args.max_number_mislabeled_training_samples,
+                max_proportion_mislabeled_training_samples=args.max_proportion_mislabeled_training_samples,
+                pcs=pop_pcs,
+                withhold_prop=args.withhold_prop,
+                test=test,
+                overwrite=overwrite,
+                only_train_on_hgdp_tgp=only_train_on_hgdp_tgp,
+                v4_population_spike=args.v4_population_spike,
+                v3_population_spike=args.v3_population_spike,
+            )
+            logger.info("Writing pop ht...")
+            pop_ht = pop_ht.checkpoint(
+                get_pop_ht(
+                    test=test, only_train_on_hgdp_tgp=only_train_on_hgdp_tgp
+                ).path,
+                overwrite=overwrite,
+                _read_if_exists=not overwrite,
+            )
+            pop_ht.transmute(
+                **{f"PC{j}": pop_ht.pca_scores[i] for i, j in enumerate(pop_pcs)}
+            ).export(
+                pop_tsv_path(test=test, only_train_on_hgdp_tgp=only_train_on_hgdp_tgp)
```

Comment:
Do we actually still need the `pop_tsv_path`?

Reply:
I'm not sure what it has been used for in the past, I can get rid of it for now